site/src/middleware/redirectToMainHost.ts

```diff
@@ -23,7 +23,14 @@ export function withRedirectToMainHostMiddleware(middleware: CustomMiddleware) {
                 return NextResponse.redirect(redirectSiteConfig.url);
             }
 
-            throw new Error(`Cannot get siteConfig for host ${host}`);
+            const response = new NextResponse(
+                '<!DOCTYPE html><html lang="en"><head><meta charset=utf-8><title>404</title></head><body>Cannot resolve domain</body></html>',
+                {
+                    status: 404,
+                },
+            );
+            response.headers.set("Content-Type", "text/html; charset=utf-8");
+            return response;
```

Comment:
```suggestion
            return NextResponse.json({ error: "Cannot resolve domain" }, { status: 404 })
```

I'd suggest returning JSON instead like we do for the site preview: https://github.com/vivid-planet/comet-starter/blob/main/site/src/middleware/sitePreview.ts#L14

Reply:
(It's easier and consistent).